File: runtimes/org.eclipse.edt.runtime.java/src/egl/lang/TypeCastException.java
/*******************************************************************************
 * Copyright (c) 2011 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package egl.lang;
import org.eclipse.edt.javart.ByteStorage;
import org.eclipse.edt.javart.Constants;
import org.eclipse.edt.javart.Program;
public class TypeCastException extends egl.lang.AnyException {
	private static final long serialVersionUID = 10L;
	@javax.xml.bind.annotation.XmlTransient
	public String castToName;
	@javax.xml.bind.annotation.XmlTransient
	public String actualTypeName;
	public TypeCastException() {
		super();
		ezeInitialize();
	}
	public void ezeCopy(Object source) {
		ezeCopy((TypeCastException) source);
	}
	public void ezeCopy(egl.lang.AnyValue source) {
		this.castToName = ((TypeCastException) source).castToName;
		this.actualTypeName = ((TypeCastException) source).actualTypeName;
	}
	public TypeCastException ezeNewValue(Object... args) {
		return new TypeCastException();
	}
	public void ezeSetEmpty() {
		castToName = Constants.EMPTY_STRING;
		actualTypeName = Constants.EMPTY_STRING;
	}
	public boolean isVariableDataLength() {
		return false;
	}
	public void loadFromBuffer(ByteStorage buffer, Program program) {
	}
	public int sizeInBytes() {
		return 0;
	}
	public void storeInBuffer(ByteStorage buffer) {
	}
	public void ezeInitialize() {
		castToName = Constants.EMPTY_STRING;
		actualTypeName = Constants.EMPTY_STRING;
	}
	public String getCastToName() {
		return (castToName);
	}
	public void setCastToName( String ezeValue ) {
		this.castToName = ezeValue;
	}
	public String getActualTypeName() {
		return (actualTypeName);
	}
	public void setActualTypeName( String ezeValue ) {
		this.actualTypeName = ezeValue;
	}
}

File: runtimes/org.eclipse.edt.runtime.javascript/runtime/eglx/lang/TypeCastException.js
/*******************************************************************************
 * Copyright © 2011, 2012 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 * IBM Corporation - initial API and implementation
 *
 *******************************************************************************/
egl.defineClass('eglx.lang', "TypeCastException", "eglx.lang", "AnyException", {
	"eze$$fileName" : "eglx/lang/Exceptions.egl",
		"constructor": function() {
			this.eze$$setInitial();
		}
		,
		"ezeCopy": function(source) {
			this.castToName = source.castToName;
			this.actualTypeName = source.actualTypeName;
		}
		,
		"eze$$setEmpty": function() {
			this.castToName = this.castToName || "";
			this.actualTypeName = this.actualTypeName || "";
		}
		,
		"eze$$setInitial": function() {
			this.eze$$setEmpty();
		}
		,
		"eze$$clone": function() {
			var ezert$$1 = this;
			var ezert$$2 = new egl.eglx.lang.TypeCastException();
			ezert$$2.castToName = ezert$$1.castToName;
			ezert$$2.actualTypeName = ezert$$1.actualTypeName;
			return ezert$$2;
		}
		,
		"eze$$getAnnotations": function() {
			if(this.annotations === undefined){
				this.annotations = {};
				this.annotations["XMLRootElement"] = new egl.eglx.xml.binding.annotation.XMLRootElement("TypeCastException", null, false);
			}
			return this.annotations;
		}
		,
		"eze$$getFieldInfos": function() {
			if(this.fieldInfos === undefined){
				var annotations;
				this.fieldInfos = new Array();
				annotations = {};
				annotations["XMLStyle"] = new egl.eglx.xml.binding.annotation.XMLElement("castToName", null, false, false);
				annotations["JsonName"] = new egl.eglx.json.JsonName("castToName");
				this.fieldInfos[0] =new egl.eglx.services.FieldInfo("castToName", "castToName", "S;", String, annotations);
				annotations = {};
				annotations["XMLStyle"] = new egl.eglx.xml.binding.annotation.XMLElement("actualTypeName", null, false, false);
				annotations["JsonName"] = new egl.eglx.json.JsonName("actualTypeName");
				this.fieldInfos[1] =new egl.eglx.services.FieldInfo("actualTypeName", "actualTypeName", "S;", String, annotations);
			}
			return this.fieldInfos;
		}
		,
		"toString": function() {
			return "[TypeCastException]";
		}
		,
		"eze$$getName": function() {
			return "eglx.lang.TypeCastException";
		}
		,
		"eze$$getChildVariables": function() {
			var eze$$parent = this;
			var childVars = this.eze$$superClass.prototype.eze$$getChildVariables.call(this);
			childVars.push({name: "castToName", value : eze$$parent.castToName, type : "eglx.lang.EString", jsName : "castToName"});
			childVars.push({name: "actualTypeName", value : eze$$parent.actualTypeName, type : "eglx.lang.EString", jsName : "actualTypeName"});
			return childVars;
		}
	}
);
